src/utils/hooks.js

```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import ThemeContext from './theme/context.js'
+
+export const useNode = (node, path, tree) => {
+  node = node || (tree && path && tree.getNode(path))
+  if (!node) throw Error(`Node not provided, and couldn't find node at ${path}`)
+  return node
+}
```

Comment:
`useNode` definitely needs to be an implementation like the one I've got WIP using the client watchNode API. If you're dropping observers and aren't triggering rerenders from the hook, this PR means we're relying on the top level observer and forcing the entire UI to rerender.

Also, this migration needs to be piecemeal since its such a huge change

Reply:
Not sure I follow. `withNode` never triggered any re-renders. Here's `contexturify`

```javascript
export let contexturify = _.flow(observer, withLoader, withNode, withTheme)
```

`observer` is applied first so any property dereferenced in the other HOCs are not tracked.

> `useNode` definitely needs to be an implementation like the one I've got WIP using the client watchNode API.

I think this comment was made with the assumption that the current `withNode` hoc re-renders components, but it does not, so this `useNode` hook is not equivalent to your watchNode API.

Reply:
> Also, this migration needs to be piecemeal since its such a huge change

It touches lots of files but they're all the same change. Perhaps we can get together to go over it.

Reply:
Ah I missed the part in the diff where you're adding observer to every component and just focused on dropping contexturify. As long as each component is still an observer, there's no issue 👍 

With that said, it does seem like we might not want to make this change given that we're going to move off observer as well with a useNode hook that also does subscription. This PR gets us closer to that end goal though, so I'm not opposed 🤔